packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -0,0 +1,244 @@
+import type {
+    AbapServiceProvider,
+    ProviderConfiguration,
+    Ui5AbapRepositoryService,
+    AxiosRequestConfig,
+    AxiosError
+} from '@sap-ux/axios-extension';
+import {
+    AbapCloudEnvironment,
+    createForAbap,
+    createForDestination,
+    createForAbapOnCloud,
+    isAxiosError
+} from '@sap-ux/axios-extension';
+import type { ServiceInfo } from '@sap-ux/btp-utils';
+import { isAppStudio, listDestinations } from '@sap-ux/btp-utils';
+import type { Logger } from '@sap-ux/logger';
+import type { BackendSystem } from '@sap-ux/store';
+import { getService, BackendSystemKey } from '@sap-ux/store';
+import { writeFileSync } from 'fs';
+import type { AbapDeployConfig, UrlAbapTarget } from '../types';
+import { getConfigForLogging, isUrlTarget } from './config';
+import { promptConfirmation, promptCredentials, promptServiceKeys } from './prompt';
+
+type BasicAuth = Required<Pick<BackendSystem, 'username' | 'password'>>;
+type ServiceAuth = Required<Pick<BackendSystem, 'serviceKeys' | 'name'>> & { refreshToken?: string };
+
+/**
+ * Check the secure storage if it has credentials for the given target.
+ *
+ * @param target - ABAP target
+ */
+export async function getCredentials<T extends BasicAuth | ServiceAuth | undefined>(
+    target: UrlAbapTarget
+): Promise<T | undefined> {
+    if (!isAppStudio()) {
+        const systemService = await getService<BackendSystem, BackendSystemKey>({ entityName: 'system' });
+        let system = await systemService.read(new BackendSystemKey({ url: target.url, client: target.client }));
+        if (!system && target.client) {
+            // check if there are credentials for the default client
+            system = await systemService.read(new BackendSystemKey({ url: target.url }));
+        }
+        return system as T;
+    } else {
+        return undefined;
+    }
+}
+
+/**
+ * Enhance axios options and create a service provided instance.
+ *
+ * @param options - predefined axios options
+ * @param target - url target configuration
+ */
+async function createAbapServiceProvider(
```

Comment:
AbapServiceProvider is not exposed. What if we need to use other services from AbapServiceProvider in the future? Do we also build it within the deploy-tooling service? An example scenario I am thinking about is one  internal GH issue that requires creating transport number dynamically during deployment.

Reply:
Good point. It really depends on the case. The functionality here is only used for deployment so far and is combining functionality of the `axios-extension` and the `store` module, so it can not be move up to either of them. If we add functionality like you describe to this module, then of course it should be reused and we could consider moving it to a submodule here that is then reused for deployment and for your use case.
For now, I would leave it as is, internal, but as soon as it is needed, we move and expose it.